docs/dev/playwright.rst

```diff
@@ -0,0 +1,96 @@
+E2E Testing with Playwright
+===========================
+
+Best practices for writing tests in Playwright
+----------------------------------------------
+
+1. **Use page objects for common interactions**:
+
+    Page objects are a design pattern that helps to abstract the details of the page structure and interactions. They
+    encapsulate the page elements and their interactions with the page. This makes the tests more readable and
+    maintainable.
+    Page objects are stored in the ``support/pageobjects`` folder. Each page object is implemented as a class containing
+    a Playwright page instance and may have instances of other page objects as well. Page object classes provide methods
+    performing common user actions or returning frequently used locators.
+    Page objects are registered as fixtures to make them easily accessible in tests without caring about their
+    initialization and teardown.
+
+2. **Use fixtures**:
+
+    Test fixture in Playwright is a setup environment that prepares the necessary conditions and state required for your
+    tests to run. It helps manage the initialization and cleanup tasks so that each test starts with a known state.
+    We use fixtures for all POMs and common test commands such as ``login``. Fixtures are defined in
+    ``support/fixtures.ts``.
+
+    To create a fixture, define its instance inside a corresponding existing type or define a new one:
+
+    .. code-block:: typescript
+
+        export type ArtemisPageObjects = {
+            loginPage: LoginPage;
+        }
+
+    2. Ensure the base test (``base``) extends the fixture type. Define a fixture with the relevant name and return the
+    desired instance as an argument of ``use()`` function as below:
+
+    .. code-block:: typescript
+
+        export const test = base.extend<ArtemisPageObjects>({
+            loginPage: async ({ page }) => new LoginPage(page)
+        });
+
+    3. Inject the fixture to tests when needed as an argument to the ``test()`` function as follows:
+
+    .. code-block:: typescript
+
+        test('Test name', async ({ fixtureName }) => {
+            // Test code
+        });
+
+3. **Use uniquely identifiable locators**:
+
+    Use unique locators to identify elements on the page. Playwright throws an error when interacting with a locator
+    that matches multiple elements on the page. To ensure uniqueness, use locators based on the element's
+    ``data-testid``, ``id``, unique ``class`` or a combination of these attributes.
+
+    Avoid using the ``nth()`` method or the ``nth-child`` selector, s they rely on the element’s position in the DOM
```

Comment:
It should be: Avoid using the ``nth()`` method or the ``nth-child`` selector, as they rely on the element’s position in the DOM